docs/hugo/content/design/ADR-2023-09-Skipping-Properties/_index.md

```diff
@@ -0,0 +1,174 @@
+[v4.Person|FullName string;FamilyName string; KnownAs string|PropertyBag PropertyBag]..[v4.Address|Label string {bg:cornsilk}]
+[v5.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress|PropertyBag PropertyBag] 
+[v5.Person]--[v5.Address|Street string; Suburb string; City string|PropertyBag PropertyBag]
+-->
+
+{{< figure src="./option-1.png" >}}
+
+We create the shape of the intermediate type by replicating the structure of the version _before_ (**v3**), essentially just copying it forward to the new version (**v4**).
+
+When we convert from **v3** to **v4**, we first convert the **v3** `Address` into a **v4** `Address` before storing it in the property bag.
+
+Then, when converting from **v4** to **v5**, we rehydrate a **v4** `Address` from the property bag and then convert that to a **v5** `Address`.
+
+The process for conversion from **v5** to **v4** to **v3** is the reverse. We convert the **v5** `Address` into a **v4** `Address` before storing it in the property bag, and then rehydrate a **v4** `Address` from the property bag and convert that to a **v3** `Address`.
+
+While this might seem to just _kick the can down the road_, our conversion framework already has support for customization of the conversions between adjacent versions of a type, allowing the mismatch between **v4** and **v5** of `Address` to be handled by augmenting the conversion.
+
+Pros:
+* Addresses the central inconsistency problem, ensuring we always serialize the same shape into intermediate property bags
+* Preserves the serialization format for existing resource versions (e.g. **v3** and **v4**)
+
+Cons:
+* Changes the object model of released resources, a breaking change for consumers of the object model
+* When there are multi-generational skips, other questions arise: Do we introduce the new type into every intermediate version, only the last, or both first and last? Each approach has advantages and drawbacks.
+
+Fatal:
+* The **v4** package already has a reference to **v5**. The Go compiler's rules prohibiting circular package dependencies preclude adding a reference to **v4** from **v5**.
+
+### Option 2: Introduce Conversion Type
+
+As for Option 1, but instead of introducing the new type into **v4** where it can't be referenced by **v5**, use a dedicated **v5/compat** subpackage.
+
+{{< figure src="./option-2.png" >}}
+
+<!-- yuml.me
+[v3.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress]
+[v3.Person]--[v3.Address|Label string]
+[v4.Person|FullName string;FamilyName string; KnownAs string|PropertyBag PropertyBag]..[v4.Address|Label string {bg:cornsilk}]
+[v5.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress|PropertyBag PropertyBag] 
+[v5.Person]--[v5.Address|Street string; Suburb string; City string|PropertyBag PropertyBag]
+-->
+
+The invariant we need to maintain here is that the **v5/compat** version of `Address` needs to have the same exact shape as the **v3** version of `Address`.
```

Comment:
Would be good to expand a bit more on how would option 2 work

Reply:
Also, would be good to have an understanding of what is `compat` subpackage

Reply:
The `compat` subpackage is literally a place to stash an extra type or two that's out of the way. Nothing more. 

I've expanded how this would work - though it's now Option 4.